Assets/PatchKit Patcher/Scripts/Utilities/GZipReadWrapperStream.cs

```diff
@@ -0,0 +1,101 @@
+using System.IO;
+using System;
+
+namespace PatchKit.Unity.Utilities
+{
+    public class GZipReadWrapperStream : Stream
+    {
+        private readonly int _minimumReadSize;
+
+        private readonly Stream _source;
+
+        public override bool CanRead
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return _source.CanSeek;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override long Length
+        {
+            get
+            {
+                return _source.Length;
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                return _source.Position;
+            }
+            set
+            {
+                _source.Position = value;
+            }
+        }
+
+        public override void Flush()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public GZipReadWrapperStream(Stream sourceStream)
+        {
+            _source = sourceStream;
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            int totalBytesRead = 0;
+            int bytesRead;
+
+            byte[] tempBuffer = new byte[count];
+
+            do
+            {
+                bytesRead = _source.Read(tempBuffer, totalBytesRead, count - totalBytesRead);
```

Comment:
@Dzejkop Is there a case when Read() may return a negative number?

Reply:
Not really, according to [MSDN](https://msdn.microsoft.com/pl-pl/library/system.io.stream.read(v=vs.110).aspx) read will return either a number of bytes read, so larger than 0 or 0 if the end of the stream has been reached.